src/main/java/chess/chessgame/ChessGame.java

```diff
@@ -0,0 +1,94 @@
+package chess.chessgame;
+
+import chess.dto.ScoreDto;
+import chess.piece.Color;
+import chess.piece.Piece;
+import chess.state.Finish;
+import chess.state.Play;
+import chess.state.Ready;
+import chess.state.State;
+import chess.utils.UnicodeConverter;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ChessGame {
+
+    private static final double MINUS_SCORE_OF_SAME_Y_PAWN = 0.5;
+
+    private final Turn turn;
+    private State state;
+
+    public ChessGame() {
+        this.state = new Ready();
+        this.turn = new Turn();
+    }
+
+    public ChessGame(String state, String turn, Map<Position, Piece> boards) {
+        this.state = computeState(state, new Chessboard(() -> boards));
```

Comment:
```suggestion
        this.state = computeState(state, new Chessboard(boards));
```
여기서는 람다를 사용할 이유가 있을까요?

Reply:
Chessboard의 생성자는 ChessboardGenerator라는 인터페이스를 전달받아 genarate함수를 통해 초기화됩니다. 이때 해당 인터페이스가 generate 함수 하나만을 가지고있기 때문에 람다를 사용했습니다! 혹시 체스보드를 직접 받는 생성자를 하나 더 추가하면 좋겠다는 의미에서 남겨주신 피드백일까요? 그렇다면 현재 존재하는 생성자로도 무리없이 생성이 가능한데 생성자를 하나 더 추가하는 것이 좋겠다고 생각하시는 이유가 궁금합니다!  🙏

Reply:
Chessboard를 사용하는 클라이언트 입장에서 생각해보면 람다를 사용해서 필요한 값을 넘겨줄거라 생각하기 어려울 것 같네요.
람다로 뭔가 계산해서 넘겨줘야하나? 라는 생각이 먼저 드네요.

Reply:
구구 말을 들으니 그렇게 보일수도 있겠네요..!!! 수정했습니다 :)